tools/Environments/DevHome.Environments/ViewModels/LandingPageViewModel.cs

```diff
@@ -157,22 +250,61 @@ public void SearchHandler(string query)
         };
     }
 
+    /// <summary>
+    /// Updates the view model to filter the compute systems according to the provider.
+    /// </summary>
+    [RelayCommand]
+    public void ProviderHandler()
+    {
+        var currentProvider = Providers[SelectedProviderIndex];
+        ComputeSystemsView.Filter = system =>
+        {
+            if (currentProvider == _stringResource.GetLocalized("AllProviders"))
+            {
+                return true;
+            }
+
+            if (system is ComputeSystemViewModel computeSystemViewModel)
+            {
+                var type = computeSystemViewModel.Type;
+                return type.Contains(currentProvider, StringComparison.OrdinalIgnoreCase);
+            }
+
+            return false;
+        };
+    }
+
     /// <summary>
     /// Updates the view model to sort the compute systems according to the sort criteria.
     /// </summary>
     [RelayCommand]
-    public void SortHandler(string critieria)
+    public void SortHandler()
     {
         ComputeSystemsView.SortDescriptions.Clear();
-        if (critieria == "Name")
-        {
-            ComputeSystemsView.SortDescriptions.Add(new SortDescription("Name", SortDirection.Ascending));
-            return;
-        }
-        else if (critieria == "Alternative Name")
+
+        // Convert to switch case
+        switch (SelectedSortIndex)

```

Comment:
Do you know if the Sort options should be sorted in any particular order? E.g ascending or do we not care? If we don't care then this works, but if we do care then we'd need to rework this to look at the selected value instead of selected index. I believe if we wanted it sorted in another language the indexes could possibly change. You'd probably have to create a list of localized strings then use the list as the data source to the combo box. 

If we don't care that the sort options aren't sorted then we're good here

Reply:
They should be sorted, but I don't understand the localization issue here. Why would the order change? Any localization change would only change texts, not their order.

Reply:
What I meant was right now we have it as 

SortByName
SortByNameDesc
SortByAltName
etc...

in the UI this looks like:

````
Name: Descending
Name: Ascending
Alternative Name: Ascending
````

In english, that isn't sorted. `Alternative Name: Ascending` would show first. In french I'm not sure if they would have the same ordering either. If we hard code the order then sure the index wouldn't change, but then that just means the text is not sorted. So I just wanted to confirm if we cared or not that the text we display to the user wouldn't be alphabetically sorted.

Reply:
That makes more sense. Yeah, we don't care about that sorting.